src/storage/index/iter.rs

```diff
@@ -0,0 +1,289 @@
+                panic!("should not happen");
+            };
+            if range.contains(&twig.key) {
+                let val = twig.get_latest_leaf();
+                if let Some(v) = val {
+                    leafs.push_back((&twig.key, &v.value, &v.version, &v.ts));
+                }
+            }
+        } else {
+            iters.push(NodeIter::new(node.iter()));
+        }
+
+        Self { iters, leafs }
+    }
+}
+
+impl<'a, P: KeyTrait + 'a, V: Clone> Iterator for IterState<'a, P, V> {
+    type Item = (Vec<u8>, &'a V, &'a u64, &'a u64);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while let Some(node) = self.iters.last_mut() {
+            let e = node.next();
+            match e {
+                None => {
+                    self.iters.pop().unwrap();
+                }
+                Some(other) => {
+                    if other.1.is_twig() {
+                        let NodeType::Twig(twig) = &other.1.node_type else {
+                            panic!("should not happen");
+                        };
+                        let val = twig.get_latest_leaf();
+                        if let Some(v) = val {
+                            self.leafs
+                                .push_back((&twig.key, &v.value, &v.version, &v.ts));
+                        }
+                        break;
+                    } else {
+                        self.iters.push(NodeIter::new(other.1.iter()));
+                    }
+                }
+            }
+        }
+
+        self.leafs
+            .pop_front()
+            .map(|leaf| (leaf.0.as_slice().to_vec(), leaf.1, leaf.2, leaf.3))
+    }
+}
+
+pub struct Range<'a, K: KeyTrait, V: Clone, R> {
+    forward: IterState<'a, K, V>,
+    range: R,
+}
+
+impl<'a, K: KeyTrait, V: Clone, R> Range<'a, K, V, R>
+where
+    K: Ord,
+    R: RangeBounds<K>,
+{
+    pub(crate) fn empty(range: R) -> Self {
+        Self {
+            forward: IterState::empty(),
+            range,
+        }
+    }
+
+    pub(crate) fn new(node: Option<&'a Arc<Node<K, V>>>, range: R) -> Self
+    where
+        R: RangeBounds<K>,
+    {
+        if let Some(node) = node {
+            Self {
+                forward: IterState::forward_scan(node, &range),
+                range,
+            }
+        } else {
+            Self {
+                forward: IterState::empty(),
+                range,
+            }
+        }
+    }
+}
+
+impl<'a, K: 'a + KeyTrait, V: Clone, R: RangeBounds<K>> Iterator for Range<'a, K, V, R> {
+    type Item = (Vec<u8>, &'a V, &'a u64, &'a u64);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while let Some(node) = self.forward.iters.last_mut() {
+            let e = node.next();
+            match e {
+                Some(other) => {
+                    if other.1.is_twig() {
+                        let NodeType::Twig(twig) = &other.1.node_type else {
+                            panic!("should not happen");
```

Comment:
`unreachable!()` is more idiomatic than `panic!("should not happen");`

Reply:
Fixed this! Thank you